### PR TITLE
v4.10 release

### PR DESCRIPTION
**Pull request by jordan-smith721**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-node/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-44054
Staging - https://deploy-preview-111--docs-pymongo.netlify.app/whats-new/#std-label-version-4.10

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/whats-new.txt [hunk 1]
```diff
@@ -15,21 +15,42 @@ What's New
...
+
+The {+driver-short+} v4.10 release includes the following new features:
+
+- Adds provisional support for a new binary BSON subtype (9), which can be used
+  for efficient storage and retrieval of :atlas:`vectors
+  </atlas-vector-search/vector-search-overview/#std-term-vector>` by using the
+  ``bson.binary.Binary.from_vector()`` and ``bson.binary.Binary.as_vector()``
+  methods. Support for BSON subtype 9 is in beta and is subject to change before the
+  generally available release.
+- Adds ``"c"`` to the ``driver.name``client metadata field when an application uses the C extension.
```
Comment (norareidy) [inline source/whats-new.txt:42]: S: add space here
```suggestion
- Adds ``"c"`` to the ``driver.name`` client metadata field when an application uses the C extension.
```

**Review by norareidy — APPROVED** [review]
LGTM + a suggestion

File: source/whats-new.txt [hunk 2]
```diff
@@ -15,21 +15,42 @@ What's New
...
+
+- Adds provisional support for a new binary BSON subtype (9), which can be used
+  for efficient storage and retrieval of :atlas:`vectors
+  </atlas-vector-search/vector-search-overview/#std-term-vector>` by using the
+  ``bson.binary.Binary.from_vector()`` and ``bson.binary.Binary.as_vector()``
+  methods. Support for BSON subtype 9 is in beta and is subject to change before the
+  generally available release.
+- Adds ``"c"`` to the ``driver.name`` client metadata field when an application uses the C extension.
+- Fixes a bug that could cause ``AsyncMongoClient`` to deadlock.
+- Fixes a bug that causes Windows to fail when importing {+driver-short+} if
```
Comment (blink1073) [inline source/whats-new.txt:44]: We can remove this, it turned out to be an incomplete fix, and was a problem with with how we were running a test in EVG.

**Comment by blink1073** [comment]
We still support MongoDB 4.0 in PyMongo 4.9 and 4.10.  MongoDB 3.6 has partial support.

File: source/whats-new.txt (hunk 2 above)
Comment (jordan-smith721) [reply source/whats-new.txt]: Removed

**Review by blink1073 — APPROVED** [review]
LGTM!

**Pull request merged**